Yosemite/Yosemite/PointOfSale/PointOfSaleCouponService.swift

```diff
@@ -10,52 +9,56 @@ import Storage
 public final class PointOfSaleCouponService: PointOfSaleItemServiceProtocol {
     private var siteID: Int64
     private let currencyFormatter: CurrencyFormatter
-    private let productsRemote: ProductsRemote
-    private let variationRemote: ProductVariationsRemoteProtocol
+    private let couponsRemote: CouponsRemote
```

Comment:
Should we use a `CouponsRemoteProtocol` here to support mocking the remote in unit tests?

Or is it preferred to mock `Network` instead in the project?

```suggestion
    private let couponsRemote: CouponsRemoteProtocol
```

Reply:
Good question, I'm not sure what we'd want to test better right now: Passing a predefined response from mocking the network, or mocking the remote logic, since a lot of it should be covered by pre-existing tests. Perhaps both! I'll take a look into it asap separately with some unit tests.